### ISSUE TITLE
Needs more GPU testing

### ISSUE THREAD

**Issue by zelikos** [zelikos/davincibox#21]
The docs are ever a work-in-progress as it is, but one big thing we're completely lacking right now is information on GPUs that work with what's provided here.

Right now, we're using AMD ROCm for AMD graphics cards. In my own testing, RX 6000 series cards---or at least, 6600 XT and up---should work OK. Other cards that are officially supported in ROCm *should* work too, but this isn't something I can verify alone. Older AMD cards are less likely to work.

For Nvidia and Intel, I can't test the former myself, and I have no idea on the latter whatsoever.

tl;dr: More testers needed with varying graphics cards

Edit:
GTX 10XX cards seem to work fine

Edit 2023-12-18:
As of ROCm 5.7.1, Resolve crashes on launch on my RX 6600 XT. Mesa's OpenCL (i.e. rusticl) lets Resolve launch successfully, but in an unusable state as video playback doesn't function and some of my projects don't load

**Comment by ahoneybun** [comment]
It seems to work fine for me on Pop!_OS 22.04. It does still seem to have the issue with certain codex but I think that's because it's the non-paid version which is Studio.

![Screenshot from 2023-10-09 11-29-14](https://github.com/zelikos/davincibox/assets/4884946/84588cd6-1504-472e-bb63-571350fc5708)

**Comment by rothgar** [comment]
I have the studio version of davinci (18.6) on bluefin and this looks like it's detecting my GTX 1070 without any issues.
![image](https://github.com/zelikos/davincibox/assets/371796/8af6660a-43eb-4364-aa10-ab89ba8150f5)

I also see resolve GPU usage from `nvidia-smi`
```
nvidia-smi
Tue Oct 10 10:17:14 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.113.01             Driver Version: 535.113.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce GTX 1070        Off | 00000000:01:00.0  On |                  N/A |
|  0%   50C    P2              36W / 151W |   3063MiB /  8192MiB |      1%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      2839      G   /usr/bin/gnome-shell                        335MiB |
|    0   N/A  N/A      3802      G   /usr/bin/Xwayland                           817MiB |
|    0   N/A  N/A    180004    C+G   /opt/resolve/bin/resolve                   1131MiB |
+---------------------------------------------------------------------------------------+
```

**Comment by ahoneybun** [comment]
@rothgar can it play MP4 video files and such?

**Comment by rothgar** [comment]
MP4 video plays just fine but aac audio doesn't work with or without distrobox. That's a licensing issue with resolve on Linux in general.

I convert all of my videos to .MOV container with PCM audio.

```
ffmpeg -i "${1}" -vcodec dnxhd -acodec:0 pcm_s16le -acodec:1 pcm_s16le -s 1920x1080 -r 30000/1001 -b:v 36M -pix_fmt yuv422p -f mov "${1%.*}".mov
```

**Comment by flexagoon** [comment]
On a laptop with Nvidia, DaVinci doesn't use the dGPU by default, and GNOME's "launch with discrete GPU" button doesn't work because it sets environment variables which don't get passed through by distrobox. The fix I found was editing the desktop file and replacing

```desktop
Exec=/usr/bin/distrobox-enter -n davincibox -- /opt/resolve/bin/resolve %u
```

with

```desktop
Exec=/usr/bin/distrobox-enter -n davincibox -- env __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia /opt/resolve/bin/resolve %u
```

**Comment by tjbbjt** [comment]
Resolve is crashing with davincibox running 18.6.4 on a Radeon VII

Also tried installing drivers:
https://repo.radeon.com/amdgpu-install/23.30.2/rhel/9.2/amdgpu-install-5.7.50702-1.el9.noarch.rpm
https://copr.fedorainfracloud.org/coprs/xxmitsu/mesa-git/

After installing either of these Resolve stopped crashing but gives an Unable to initialize OpenGL error

![opengl error](https://github.com/zelikos/davincibox/assets/102318757/7523ba66-f191-4088-9493-4bf77b12bace)

**Comment by trafotin** [comment]
![Screenshot from 2023-12-31 21-11-43](https://github.com/zelikos/davincibox/assets/108948841/7144ee53-8c4e-4584-846a-1a9ba3e03b8a)
If you use a Nvidia GPU, add the flag "--nvidia" when manually building in Distrobox. I'm testing with a 4060M, on Fedora with  Wayland/GNOME.

Getting past the "make your sure no displays are plugged into your discrete GPU" requires you run DaVinci in X11 first with the Nvidia drivers on your host, then open it in Wayland and then it works.

After this point, the above warning shows because there's somewhere in the pipeline where switcheroo can't give Podman access to a dGPU. You can still make edits, but nothing plays.

Running natively on Fedora, Resolve works, so it's not an issue with the container or its build quality as much as it is discrete Nvidia GPUs not playing ball.

**Comment by flexagoon** [comment]
@trafotin have you tried my solution?

https://github.com/zelikos/davincibox/issues/21#issuecomment-1827508764

This worked for me even without the `--nvidia` option

**Comment by trafotin** [comment]
> @trafotin have you tried my solution?

We're further along, but I'm still getting crashes. Crash is identical on both Wayland and X11.
```
ActCCMessage Already in Table: Code= c005, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= c006, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= c007, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= 2282, Mode=  0, Level=  0, CmdKey= 8, Option= 0
PnlMsgActionStringAdapter Already in Table: Code= 615e, Mode=  0, Level=  0, CmdKey= -1, Option= 0
18.6.0.0009 Linux/Clang x86_64
Main thread starts: 0E20E000
Failed to load config file "%u".Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
resolve: /home/jenkins/jenkins/workspace/resolve/Resolve/Cyclone/Main/AppConfig.cpp:296: void AppConfig::LoadAllSiteInfo(): Assertion `m_SiteEnabledIdx > 0' failed.
log4cxx: No appender could be found for logger (Main).
log4cxx: Please initialize the log4cxx system properly.


==========[CRASH DUMP]==========
#TIME Mon Jan  1 15:50:08 2024 - Uptime 00:00:00 (hh:mm:ss)
#PROGRAM_NAME DaVinci Resolve v18.6.0.0009 (Linux/Clang x86_64)
#BMD_ARCHITECTURE x86_64
#BMD_BUILD_UUID 445d7d93-25e8-4e6d-a54e-1bb3ee745c58
#BMD_GIT_COMMIT 2541dac8b8ea4bd08de600b58edcb7d95aa657bc
#BMD_UTIL_VERSION 18.6.0.0009
#OS Linux

/opt/resolve/bin/resolve() [0x48069c9]
/opt/resolve/bin/resolve() [0x24c8b82]
/lib64/libc.so.6(+0x3e9a0) [0x7f5e0f25c9a0]
/lib64/libc.so.6(+0x90834) [0x7f5e0f2ae834]
/lib64/libc.so.6(raise+0x1e) [0x7f5e0f25c8ee]
/lib64/libc.so.6(abort+0xdf) [0x7f5e0f2448ff]
/lib64/libc.so.6(+0x2681b) [0x7f5e0f24481b]
/lib64/libc.so.6(+0x36c57) [0x7f5e0f254c57]
/opt/resolve/bin/resolve() [0x24f9dec]
/opt/resolve/bin/resolve() [0x24f6f53]
/opt/resolve/bin/resolve() [0x24f68a8]
/opt/resolve/bin/resolve() [0x40df168]
/opt/resolve/bin/resolve() [0x3ac4196]
/opt/resolve/bin/resolve() [0x3ac3fce]
/opt/resolve/bin/resolve() [0x251cbc8]
/opt/resolve/bin/resolve() [0x251cad1]
/opt/resolve/bin/resolve() [0x24c6740]
/opt/resolve/bin/resolve() [0x24c15fe]
/lib64/libc.so.6(+0x2814a) [0x7f5e0f24614a]
/lib64/libc.so.6(__libc_start_main+0x8b) [0x7f5e0f24620b]
/opt/resolve/bin/resolve() [0x24c024b]
Signal Number = 6

================================
resolve: /home/jenkins/jenkins/workspace/resolve/Resolve/Cyclone/Main/AppConfig.cpp:296: void AppConfig::LoadAllSiteInfo(): Assertion `m_SiteEnabledIdx > 0' failed.


==========[CRASH DUMP]==========
#TIME Mon Jan  1 15:50:08 2024 - Uptime 00:00:00 (hh:mm:ss)
#PROGRAM_NAME DaVinci Resolve v18.6.0.0009 (Linux/Clang x86_64)
#BMD_ARCHITECTURE x86_64
#BMD_BUILD_UUID 445d7d93-25e8-4e6d-a54e-1bb3ee745c58
#BMD_GIT_COMMIT 2541dac8b8ea4bd08de600b58edcb7d95aa657bc
#BMD_UTIL_VERSION 18.6.0.0009
#OS Linux

/opt/resolve/bin/resolve() [0x48069c9]
/opt/resolve/bin/resolve() [0x24c8b82]
/lib64/libc.so.6(+0x3e9a0) [0x7f5e0f25c9a0]
/lib64/libc.so.6(+0x90834) [0x7f5e0f2ae834]
/lib64/libc.so.6(raise+0x1e) [0x7f5e0f25c8ee]
/lib64/libc.so.6(abort+0xdf) [0x7f5e0f2448ff]
/lib64/libc.so.6(+0x2681b) [0x7f5e0f24481b]
/lib64/libc.so.6(+0x36c57) [0x7f5e0f254c57]
/opt/resolve/bin/resolve() [0x24f9dec]
/opt/resolve/bin/resolve() [0x24f6f53]
/opt/resolve/bin/resolve() [0x24f68a8]
/opt/resolve/bin/resolve() [0x250d8f9]
/opt/resolve/bin/resolve() [0x24c8d93]
/lib64/libc.so.6(+0x3e9a0) [0x7f5e0f25c9a0]
/lib64/libc.so.6(+0x90834) [0x7f5e0f2ae834]
/lib64/libc.so.6(raise+0x1e) [0x7f5e0f25c8ee]
/lib64/libc.so.6(abort+0xdf) [0x7f5e0f2448ff]
/lib64/libc.so.6(+0x2681b) [0x7f5e0f24481b]
/lib64/libc.so.6(+0x36c57) [0x7f5e0f254c57]
/opt/resolve/bin/resolve() [0x24f9dec]
/opt/resolve/bin/resolve() [0x24f6f53]
/opt/resolve/bin/resolve() [0x24f68a8]
/opt/resolve/bin/resolve() [0x40df168]
/opt/resolve/bin/resolve() [0x3ac4196]
/opt/resolve/bin/resolve() [0x3ac3fce]
/opt/resolve/bin/resolve() [0x251cbc8]
/opt/resolve/bin/resolve() [0x251cad1]
/opt/resolve/bin/resolve() [0x24c6740]
/opt/resolve/bin/resolve() [0x24c15fe]
/lib64/libc.so.6(+0x2814a) [0x7f5e0f24614a]
/lib64/libc.so.6(__libc_start_main+0x8b) [0x7f5e0f24620b]
/opt/resolve/bin/resolve() [0x24c024b]
Signal Number = 6

================================
```
I haven't tried without the Nvidia option, but will respond to this post and tag you when I do.

**Comment by tjbbjt** [comment]
Was able to get an Arch Linux distrobox working, the amdgpu-pro-oglp using progl prefix would crash but opencl-amd and mesa-rusticl-git with "export OCL_ICD_VENDORS= export OCL_ICD_FILENAMES=/usr/lib64/libRusticlOpenCL.so export RUSTICL_ENABLE=radeonsi" works. 

So Mesa Rusticl 24 should fix media not playing in Fedora.

**Comment by theonekingboo** [comment]
the software is fully functional on my GTX 760 card
![image](https://github.com/zelikos/davincibox/assets/139255943/a811161c-3744-4fa0-b26a-5607d4668a84)

**Comment by shubhisroking** [comment]
![image](https://github.com/zelikos/davincibox/assets/138035866/1a1ec96a-f366-4d92-8c58-3524c7f77788)
I am having same issue as @trafotin where nothing plays on an NVIDIA laptop, local installation seems to work fine

**Comment by shubhisroking** [comment]
> @shubhisroking what does this command output for you?
> 
> ```shell
> podman run \
>     --user 1000:1000 \
>     --security-opt=no-new-privileges \
>     --cap-drop=ALL \
>     --security-opt label=type:nvidia_container_t  \
>     --device=nvidia.com/gpu=all \
>     docker.io/nvidia/samples:vectoradd-cuda11.2.1
> ```
> 
> I am using uBlue, which has [support for nvidia in containers](https://universal-blue.org/images/nvidia/#using-nvidia-gpus-in-containers) out of the box. If the above command doesn't work and you tried building the distrobox both with the `--nvidia` flag and without it, try setting nvidia container runtime up.

![2024-01-24_10-57](https://github.com/zelikos/davincibox/assets/138035866/5653cbe6-91a0-44fc-a74b-9d0d8067c167)
seems to run fine after setting up nvidia container toolkit, but still nothing plays in davinci resolve even, tested with and without --nvidia option

**Comment by flexagoon** [comment]
@shubhisroking weird. Can you provide Davinci logs please?

**Comment by shubhisroking** [comment]
> @shubhisroking weird. Can you provide Davinci logs please?

Hey I just did a reboot and it works fine now 
![2024-01-24_13-20](https://github.com/zelikos/davincibox/assets/138035866/1c127a25-342e-407a-a3bd-9ed1b879c968)

**Comment by aldecrat** [comment]
![image](https://github.com/zelikos/davincibox/assets/158589195/cf0a5658-8945-494f-b16b-f9788ed79f46)
![image](https://github.com/zelikos/davincibox/assets/158589195/ad26e753-d69b-4381-a18f-5985ee9b5864)

Using Radeon RX 5500XT. Installation worked but on startup the graphics card does not appear in the list. Any help would be welcome. I also haven't a baldy what I am doing and my son is in despair.
Thanks.

**Comment by trafotin** [comment]
> Using Radeon RX 5500XT. Installation worked but on startup the graphics card does not appear in the list. Any help would be welcome. I also haven't a baldy what I am doing and my son is in despair... and so am I. Thanks.

If you are on Wayland (guessing because no titlebars), try start Resolve in X11 once, then start it up again in Wayland. Although I wonder if that is a thing with AMD users.

There's also the possibility your GPU isn't being seen by container. Are you using a discrete AMD card?

**Comment by aldecrat** [comment]
I am using X11 on Debian testing with XFCE. Yeah, I am using a discrete AMD card.
My son says that the GPU is not being seen by the container and CLinfo.
Thanks.

**Comment by tjbbjt** [comment]
> I am using X11 on Debian testing with XFCE. Yeah, I am using a discrete AMD card. My son says that the GPU is not being seen by the container and CLinfo. Thanks.

After entering the davincibox you could try

`export OCL_ICD_VENDORS= export OCL_ICD_FILENAMES=/usr/lib64/libRusticlOpenCL.so export RUSTICL_ENABLE=radeonsi`

and then 

`/opt/resolve/bin/resolve`

This will more than likely allow Resolve to start but no media will decode. Personally I could not get davincibox, flatpak or any flavor of Linux distrobox working with Resolve and my Radeon VII other than Arch Linux. I wrote out steps on how to get that working in another forum [here](https://forums.opensuse.org/t/davinci-resolve-on-tumbleweed-with-amd/171524/11). Hope any of this helps!

**Comment by aldecrat** [comment]
Thanks very much for your efforts. It appears that Davinci is not really ready for Linux yet.

**Comment by flexagoon** [comment]
@aldecrat davinci is very much ready for Linux, it just isn't intended to be run inside of a container

It also generally just works better with Nvidia GPUs

**Comment by theonekingboo** [comment]
@aldecrat i got this answer from reddit : 
```
Sharing my experience on Arch. Up to DR 18.1.1, Mesa 22 and ROCm OpenCL 5.3.3 (installed from CN repo, as 5.4 got adopted into official repo) I was able to use DR just fine, including Fusion. Starting from DR 18.1.2 and later versions of ROCm OpenCL I noticed crashes as you described.

Last night I decided to use both AMDGPU-PRO OpenGL and OpenCL packages (meaning, all-proprietary graphic components) and upgrade to DR 18.5. It run, Fusion works! Downside was it's took soooooo long for preview to cached up and display.

I should try again with Mesa OpenGL, cause I don't feel comfortable using proprietary OpenGL. If Mesa OpenGL works with opencl-amd then maybe something went wrong with rocm-opencl-runtime packaging on Arch repo.
```

**Comment by randogoth** [comment]
Works on my Gigabyte Aero 15 X-9 running Bluefin (F39). I had the black screen issue (#65) and had to [edit the desktop file](https://github.com/zelikos/davincibox/issues/21#issuecomment-1827508764). 

Launching the app froze at the splash screen until I rebooted.

![image](https://github.com/zelikos/davincibox/assets/4802166/1a6745de-c4c0-47e0-b3c0-1eda63945a77)

### nvidia-smi
```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.78                 Driver Version: 550.78         CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 2070 ...    Off |   00000000:01:00.0  On |                  N/A |
| N/A   53C    P8              8W /   80W |      28MiB /   8192MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      5593      G   /usr/bin/gnome-shell                            1MiB |
+-----------------------------------------------------------------------------------------+
```

**Comment by tmarplatt** [comment]
Host: Mint Cinnamon 21.3 (based on Ubuntu jammy)
GPU: RX 5500 XT

First off, Resolve doesn't launch without elevated privileges, so I need:

`distrobox enter -n davincibox -- sudo /usr/bin/run-davinci`

Unfortunately, same issue as @aldecrat Resolve can't see my GPU, the GPU list is empty.

I saw the `amdgpu` kernel module is loaded in the container, but I'm not sure how to verify OpenGL is working as I'm no Fedora user. For some reason there is no `glxinfo` available on $PATH, even though mesa seems to be installed. Or how to verify the container system sees my GPU. Would appreciate some help here.

**Comment by user7z** [comment]
i have and intel iGPUs , iam welcome to help test this project

**Comment by badlydrawnface** [comment]
My RX580 was detected, but it used OpenCL, and video playback did not seem to be working. Resolve seemed to only be using about 45-50 MB of RAM.

OS: CachyOS with this container

Might be time for a new GPU soon, the 580 is still a sleeper even in 2024 tho...
![image](https://github.com/zelikos/davincibox/assets/53501005/b39b6211-e7fc-41f6-8ea0-160beebb949a)

**Comment by ricardopiloto** [comment]
Host: Fedora 40
GPU: RX 6750 XT

Managed to install without any problems but whenever I try to add a video to the media pool I get an error regarding GPU processing (error -57). Tried second option with rusticl but got the same result only difference being that with rusticl only one GPU is shown.

![image](https://github.com/zelikos/davincibox/assets/13372861/af70a281-99e4-43a5-8351-284f19cafd64)

![Screenshot_20240528_132420](https://github.com/zelikos/davincibox/assets/13372861/2c3e867a-f9fb-42e1-81dd-9d99c164f06b)

**Comment by zelikos** [comment]
@ricardopiloto Can confirm the same issue on my end; neither ROCm nor rusticl working with Resolve right now.

**Comment by BluedragonMask** [comment]
![image](https://github.com/zelikos/davincibox/assets/101012161/b19cebee-e814-405a-b802-6f440d8bb16a)
Yup, same issue here. I thought I was going crazy till I scrolled down to the end of this

**Comment by ItsRogueRen** [comment]
I'm having the same GPU error on my RX 7700XT, I thought I did something horribly wrong lol.

At this point I assume this problem is on Black Magic to fix? Or is it possible to be done by this project?

**Comment by zelikos** [comment]
I'll be closing this issue.

When this project was just beginning, having a "call to action" issue thread like this to have various GPUs tested was great, and I want to thank each and everyone who reported test results here. I think we've determined that Nvidia generally seems to work, AMD has its issues, and...Intel is somewhat unknown, but we have a [separate issue](#71) for Intel now anyway.

That being said, we're at a point where it'd be better to have separate issues opened for specific problems, to make sure things don't get lost in the shuffle (such as https://github.com/zelikos/davincibox/issues/21#issuecomment-2106571189, which sounds like a similar issue as #76?).

If you've posted about a problem in this thread that doesn't already have an open issue report, please open a new issue about it.
